tests/test_import_file.py

```diff
@@ -0,0 +1,176 @@
+    # date below the constraint we set (20 years before today date)
+    ("01-01-2000", False),
+    # date in the range we set in the function
+    ("01-01-2019", 5, True),
+    # date below the range we set in the function
+    ("01-01-2017", 2, False)
+]
+
+check_validity_of_text_tests = [
+    (r"Head1, Content1, 05-21-2019", True),
+    (r"Head1Content1, 05-21-2019", False),
+    # title can't be empty
+    (r"  , Content1, 05-21-2019", False),
+    # content may be empty
+    (r"Head1, , 05-21-2019", True),
+    # date can't be empty
+    (r"Head1, Content1, ", False),
+    # row cant have multiple events
+    (r"Head1, Content1, 05-21-2019, Head2, Content2, 05-21-2019", False)
+]
+
+import_txt_file_tests = [
+    # txt file
+    (r"tests/files_for_import_file_tests/sample_calendar_data_to_import_1.txt",
+        [
+            {'Head': 'Head1', 'Content': 'Content1',
+             'Date': datetime.datetime(2019, 5, 21, 0, 0)},
+            {'Head': 'Head2', 'Content': 'Content2',
+             'Date': datetime.datetime(2010, 1, 11, 0, 0)},
+            {'Head': 'Head3', 'Content': 'Content3',
+             'Date': datetime.datetime(2022, 2, 2, 0, 0)}
+        ]),
+    # csv file
+    (r"tests/files_for_import_file_tests/sample_calendar_data_to_import_1.csv",
+        [
+            {'Head': 'Head1', 'Content': 'Content1',
+             'Date': datetime.datetime(2019, 5, 21, 0, 0)},
+            {'Head': 'Head2', 'Content': 'Content2',
+             'Date': datetime.datetime(2010, 1, 11, 0, 0)},
+            {'Head': 'Head3', 'Content': 'Content3',
+             'Date': datetime.datetime(2022, 2, 2, 0, 0)}
+        ])
+]
+
+import_ics_tests = [
+    # ics file
+    (r"tests/files_for_import_file_tests/sample.ics",
+        [
+            {'Head': 'HeadA', 'Content': 'Content1',
+             'Date': datetime.datetime(2019, 8, 2, 10, 34)},
+            {'Head': 'HeadB', 'Content': 'Content2',
+             'Date': datetime.datetime(2019, 8, 2, 20, 0)}
+        ]),
+    # ics file
+    (r"tests/files_for_import_file_tests/sample2.ics", [])
+]
+
+before_import_checking_tests = [
+    (r"tests/files_for_import_file_tests/sample_calendar_data_to_import_1.txt",
+     True),
+    (r"tests/files_for_import_file_tests/sample_above_5mb.txt", False),
+    (r"tests/files_for_import_file_tests/sample.blabla", False),
+    (r"tests/files_for_import_file_tests/sample2.blabla", False)
+]
+
+
+user_click_import_tests = [
+    (r"tests/files_for_import_file_tests/sample_calendar_data_to_import_1.txt",
+     1, "Import success"),
+    (r"tests/files_for_import_file_tests/sample.ics", 1,
+     "Import success"),
+    (r"tests/files_for_import_file_tests/sample_above_5mb.txt", 1,
+     "Import failed"),
+    (r"tests/files_for_import_file_tests/sample.blabla", 1,
+     "Import failed"),
+    (r"tests/files_for_import_file_tests/sample_same_date_many_times.txt", 1,
+     "Import failed")
+]
+
+
+@pytest.mark.parametrize("file", file_size_tests)
```

Comment:
For this and the rest of the tests using the @pytest.mark.parametrize decorator, it would be better to place all the variables you use here, instead of calling it "file" then accessing it by index. Makes it also easier to understand the code without needing to jump between different parts of the file.

Reply:
Done